src/libData/AccountData/AccountStore.cpp

```diff
@@ -146,756 +97,134 @@ int AccountStore::Deserialize(const vector<unsigned char>& src,
-        }
+        unsigned int curOffset = offset;
+        uint256_t totalNumOfAccounts
+            = GetNumber<uint256_t>(src, curOffset, UINT256_SIZE);
+        LOG_GENERAL(INFO,
+                    "Total Number of Accounts Delta: " << totalNumOfAccounts);
+        curOffset += UINT256_SIZE;
 
-        if (!TransferBalance(m_curContractAddr, toAddr, amount))
-        {
-            return false;
-        }
-        // what if the positive value is come from a failed function call
-        if (deducted > 0)
+        Address address;
+        Account account;
+        unsigned int numberOfAccountDeserialze = 0;
+        while (numberOfAccountDeserialze < totalNumOfAccounts)
         {
-            if (!IncreaseBalance(m_curContractAddr, deducted))
-            {
-                return false;
-            }
-        }
-        IncreaseNonce(m_curContractAddr);
-
-        return true;
-    }
-    else
-    {
-        LOG_GENERAL(INFO, "Call another contract");
+            numberOfAccountDeserialze++;
 
-        Address toAddr;
-        Json::Value params;
-        for (auto& p : json["message"]["params"])
-        {
-            if (p["vname"].asString() == "to"
-                && p["type"].asString() == "Address")
-            {
-                toAddr = Address(p["value"].asString());
-            }
-            else
-            {
-                params.append(p);
-            }
-        }
-        Json::Value p_sender;
-        p_sender["vname"] = "sender";
-        p_sender["type"] = "Address";
-        p_sender["value"] = "0x" + m_curContractAddr.hex();
-        params.append(p_sender);
-
-        // if (params.empty())
-        // {
-        //     params = Json::arrayValue;
-        // }
-
-        Account* toAccount = GetAccount(toAddr);
-        if (toAccount == nullptr)
-        {
-            LOG_GENERAL(WARNING, "The target contract account doesn't exist");
-            return false;
-        }
+            // Deserialize address
+            copy(src.begin() + curOffset,
+                 src.begin() + curOffset + ACC_ADDR_SIZE,
+                 address.asArray().begin());
+            curOffset += ACC_ADDR_SIZE;
 
-        if (ExportCallContractFiles(
-                toAccount,
-                CompositeContractData(json["message"]["_tag"].asString(),
-                                      json["message"]["_amount"].asString(),
-                                      params)))
-        {
-            Address t_address = m_curContractAddr;
-            m_curContractAddr = toAddr;
-            if (!SysCommand::ExecuteCmdWithoutOutput(GetCallContractCmdStr()))
-            {
-                LOG_GENERAL(WARNING,
-                            "Interpreter does not admit this contract chain "
-                            "invocation");
-                return false;
-            }
-            if (ParseCallContractOutput())
+            // Deserialize accountDelta
+            Account* oriAccount = GetAccount(address);
```

Comment:
Can we avoid raw ptr here?

Reply:
In AccountStoreBase.h:36, m_addressToAccount is stored in shared_ptr<MAP>.
This issue should not be a problem.